### DISCUSSION TITLE
how to gain multipiler after optimizer

### DISCUSSION THREAD

**Discussion by PanCongcong1997** [yalmip/YALMIP#954] [Optimization modelling] [answered]
Hello, John,

Why the multipiler after "optimizer" is NAN? And can you tell me how to gain the multipilers after "optimizer" in  below  situation?

%%Constraints
SP_cst=...

%% objective
SP_obj=Prob_Sce(i_Sce)*sum(sum(P_loadsheds))*S_base*cost_loadshed;%
ops=sdpsettings('solver','Gurobi','verbose',2,'gurobi.mipgap',0.5/100);%0.5%
sol=optimizer(SP_cst,SP_obj,ops,[x_unit;x_line;P_windSpeed;afitall_SP;bfitall_SP],[P_loadsheds]);

%% for all scenarios
P_loadshedsk=[];
for i_Sce =1:2%NS
       x_unit=Sce(1:num_unit,i_Sce);
       x_line=Sce(num_unit+1:end,i_Sce);
       P_windSpeed=P_windSce(i_Sce,:)';%1*24
       ...
       tmpsol=sol{[x_unit;x_line;P_windSpeed;afitall_SP;bfitall_SP]};%solve
       _**lamdak1=dual(SP_cst('special2'))**_
       P_loadshedsk=[P_loadshedsk; sum(value(tmpsol))];%p.u.
end

The output are as below:
Solved in 109 iterations and 0.08 seconds
Optimal objective  2.680138913e+03
lamdak1 =
   NaN   NaN   NaN   ...

**Comment by johanlofberg** [comment, marked as answer]
You cannot use dual (in the same way you cannot use value) when using optimizer. The 5th output from the sol is the duals, but it is up to you to understand how they connect to your high-level model (and that is typically not trivial but has to be done by reverse engineering and experimentation)

**Reply by PanCongcong1997** [reply]
I got it. Thanks, John.
